scripts/hist_importer.py

```diff
@@ -0,0 +1,158 @@
+# Copyright 2014-2017 Josefson Souza <josefson.br@gmail.com>
+
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+
+"""Tool to import browser history from other browsers."""
+
+
+import argparse
+import sqlite3
+import sys
+import os
+
+
+def parse():
+    """Parse command line arguments."""
+    description = ("This program is meant to extract browser history from your"
+                   "previous browser and import them into qutebrowser.")
+    epilog = ("Databases:\n\tQutebrowser: Is named 'history.sqlite' and can be"
+              " found at your --basedir. In order to find where your basedir"
+              " is you can run ':open qute:version' inside qutebrowser."
+              "\n\tFirerox: Is named 'places.sqlite', and can be found at your"
+              "system\"s profile folder. Check this link for where it is locat"
+              "ed: http://kb.mozillazine.org/Profile_folder"
+              "\n\tChrome: Is named 'History', and can be found at the respec"
+              "tive User Data Directory. Check this link for where it is locat"
+              "ed: https://chromium.googlesource.com/chromium/src/+/master/"
+              "docs/user_data_dir.md\n\n"
+              "Example: hist_importer.py -b firefox -s /Firefox/Profile/"
+              "places.sqlite -d /qutebrowser/data/history.sqlite")
+    parser = argparse.ArgumentParser(
+        description=description, epilog=epilog,
+        formatter_class=argparse.RawTextHelpFormatter
+    )
+    parser.add_argument('-b', '--browser', dest='browser', required=True,
+                        type=str, help='Browsers: {firefox, chrome}')
+    parser.add_argument('-s', '--source', dest='source', required=True,
+                        type=str, help='Source: Full path to the sqlite data'
+                        'base file from the source browser.')
+    parser.add_argument('-d', '--dest', dest='dest', required=True, type=str,
+                        help='Destination: The full path to the qutebrowser '
+                        'sqlite database')
+    return parser.parse_args()
+
+
+def open_db(data_base):
+    """Open connection with database."""
+    if os.path.isfile(data_base):
+        conn = sqlite3.connect(data_base)
+        return conn
+    else:
+        raise sys.exit('\nDataBaseNotFound: There was some error trying to to'
```

Comment:
Why the `\n`, and why `DataBaseNotFound:`?

Reply:
If the user pass a file that does not exists by mistake or whatever, sqlite3.connect() will create an empty database file there. So in order to prevent the script from creating those empty databases i put that ```if os.path.isfile(data_base):```.
In case the database does not exist, it won't create an empty file at the user config directory.

Reply:
Yeah, the check is fine - but the message seems confusing. I don't think the `DataBaseNotFound:` prefix is needed, and I wonder what the initial `\n` is for.